demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/NotificationReceiver.java

```diff
@@ -0,0 +1,14 @@
+package com.novoda.downloadmanager.demo.extended;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+public class NotificationReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        context.startActivity(new Intent(context, MainActivity.class).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK));
```

Comment:
The action isn't hidden (because you still need it to do checks like the one you mention), just there is a convenience method now